doc-checking-app-api/src/main/java/uk/gov/di/authentication/app/lambda/DocAppCallbackHandler.java

```diff
@@ -287,4 +289,51 @@ private APIGatewayProxyResponseEvent redirectToFrontendErrorPage() {
                                 .toString()),
                 null);
     }
+
+    private APIGatewayProxyResponseEvent generateNoSessionErrorResponse(
+            Map<String, String> queryStringParameters)
+            throws ParseException, DocAppCallbackException {
+        LOG.info(
+                "Attempting to generate error response using state. CustomDocAppClaimEnabled: {}",
+                configurationService.isCustomDocAppClaimEnabled());
+        if (configurationService.isCustomDocAppClaimEnabled()
+                && queryStringParameters.containsKey("error")
+                && queryStringParameters.get("error").equals(OAuth2Error.ACCESS_DENIED.getCode())
+                && queryStringParameters.containsKey("state")) {
```

Comment:
I'd suggest pulling this check into a separate method for readability.

Reply:
Good shout. I have updated